proposals/0000-scan-takewhile-dropwhile.md

```diff
@@ -0,0 +1,138 @@
+# Feature name
+
+* Proposal: TBD
+* Author(s): [Kevin Ballard](https://github.com/kballard)
+* Status: **Awaiting review**
+* Review manager: TBD
+
+## Introduction
+
+Add 3 new `SequenceType` functions `scan(_:combine:)`, `takeWhile(_:)`, and
+`dropWhile(_:)`, with overrides as appropriate on `CollectionType`,
+`LazySequenceType`, and `LazyCollectionType`, as well as a global function
+`iterate(_:apply:)`.
+
+Swift-evolution thread:
+[Proposal: Add scan, takeWhile, dropWhile, and iterate to the stdlib](https://lists.swift.org/pipermail/swift-evolution/Week-of-Mon-20151228/004690.html)
```

Comment:
I’d like to encourage people to use gmane links (in this case, http://news.gmane.org/find-root.php?message_id=1451347170.2635180.478054954.2AD975B7%40webmail.messagingengine.com) because they are so much more useful.